Comments for En_Fr Frogs (do not merge)

engineer124 opened the pull request.

fig02 commented:
Great job on your first actor!
Im not going to be able to review this until later tonight or possibly tomorrow. But after scrolling through I do want to mention that I think there are alot of comments that I think can be removed.

Documentation is definitely a good thing, but there are plenty of things that are very easy to tell just by reading the code/variable names. We are aiming for a very self documenting codebase that is very easy to read and make sense of. There are definitely times when comments are needed to go into more detail, but you dont need to annotate alot of things.

I can grab some examples later if you like, but just thought id mention that now.

And to that end, we dont want comments for the text ids, it will get cluttered very fast. the text table will be in the repo at a later time for easy looking up

engineer124 commented:
Thanks Fig! 
It was a ton of fun trying to piece together frogs. Yeah I'm happy to take away some of the comments, they were mostly beneficial for me anyway to figure out the flow of the actor. 
I also realized I never updated the data names and switched them to static. 
I didn't delete any files because I wasn't sure which files need to be deleted. Can I remove all func assembly files where the functions have been decomped?
I was also hesitant to rename functions that were called by EnFr_Update (the one decomp I couldn't work out) since it wouldn't let me do the ./diff.py properly if I renamed those certain functions. But I put my idea for a name beside the functions.

fig02 commented:
As far as the diff goes, all you need to do is update your expected folder after renaming. make sure your build is OK before recopying the spec.
And yes all asm can be removed that isnt needed. You do `./tools/fin_unused_asm.sh -d` to quickly do this

engineer124 commented:
Okay sweet. I'll update my PR in the next few hours

petrie911 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+    u8 songIndex;
+
+    if (this->jumpCounter < 48) {
+        if (!(this->jumpCounter & 3)) { // % 4
```
```suggestion
        if ((this->jumpCounter % 4) == 0 ) { // % 4
```
assuming it matches

engineer124 replied:
It sure does

petrie911 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+}
+
+void EnFr_ChildSongFirstTime(EnFr* this, GlobalContext* globalCtx) {
+    EnFr* frog;
+    frog = sEnFrPointers.frogs[sSongToFrog[this->songIndex]];
+    if (frog->isActive == false) {
```
```suggestion
    EnFr* frog= sEnFrPointers.frogs[sSongToFrog[this->songIndex]];
    
    if (frog->isActive == false) {
```

petrie911 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+void EnFr_Deactivate(EnFr* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BD94.s")
+    EnFr* frog_loop1;
+    EnFr* frog_loop2;
```
do these need separate temps?

engineer124 replied:
It doesn't match if there isn't two temps. a1, v0, v1 all get swapped if only 1 temp is used. Although there might be a solution with 1 temp, I just didn't find it.

petrie911 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BF10.s")
+        frog_loop1 = sEnFrPointers.frogs[frogIndex];
+        // Translation: There are no frogs!?
+        if (!frog_loop1) {
```
```suggestion
        if (frog_loop1 == NULL) {
```
Pointers should be compared to NULL. There's a few more of these in the file.

petrie911 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C524.s")
+s32 EnFr_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
```
```suggestion
s32 EnFr_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx,
```

petrie911 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C580.s")
+void EnFr_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
```
```suggestion
void EnFr_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
```

petrie911 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+        func_800D1FD4(&globalCtx->mf_11DA0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C830.s")
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_fr.c", 0x6CA),
```
```suggestion
        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_fr.c", 1738),
```

petrie911 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+    gSPSegment(POLY_OPA_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(sDLists[this->eyeTexIndex]));
+    gSPSegment(POLY_OPA_DISP++, 0x09, SEGMENTED_TO_VIRTUAL(sDLists[this->eyeTexIndex]));
+    SkelAnime_DrawFlexOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount,
+                          EnFr_OverrideLimbDraw, EnFr_PostLimbDraw, &this->actor);
```
```suggestion
                          EnFr_OverrideLimbDraw, EnFr_PostLimbDraw, this);
```

petrie911 reviewed (commented):
Some stylistic comments and a few other things.

engineer124 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -8,14 +9,169 @@ void EnFr_Init(Actor* thisx, GlobalContext* globalCtx);
+
+// u8 Frog Index
+// Frog to Index for Song Flag (sSongIndex) Mapping
+u8 sFrogToSongIndex[] = { 0x02,   // Frog 0: (Yellow - Middle)
+                          0x03,   // Frog 1: (Blue - Front Left)
+                          0x04,   // Frog 2: (Red - Front Right)
+                          0x00,   // Frog 3: (Purple - Back Left)
+                          0x01 }; // Frog 4: (White - Back Right)
```
The above suggestions all make sense. I thought about doing an enum for frogs, but after trying it I realized there were not many places where the enum were used. Basically this one array and one special case later telling the game to not shrink the purple frog. Most of the time the frogs are all accessed through a for-loop. Would you still suggest an enum?

engineer124 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+    this->actor.draw = EnFr_Draw;
+}
 
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/EnFr_Update.s")
```
I'll have to remind myself how those preprocessors work in C, but yes I can change that

engineer124 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+
+    if (this->blinkTimer) {
+        this->blinkTimer--;
+    }
+
+    else if (this->eyeTexIndex) {
+        this->eyeTexIndex = 0;
+        this->blinkTimer = (s16)(Math_Rand_ZeroFloat(60.0f) + 20.0f);
+        this->blinkFunc = EnFr_DecrementBlinkTimer;
+    }
+
+    else {
```
Yes all my code was guilty of this. I'm being more careful now. Same applies to 'else if' I'm assuming?

engineer124 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+}
+
+void EnFr_InitJumpingOutOfWater(EnFr* this, GlobalContext* globalCtx) {
+    AnimationHeader* objSegChangeAnime = &D_060007BC;
```
No as it turns out!

engineer124 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+
+void EnFr_SetScaleActive(EnFr* this, GlobalContext* globalCtx) {
+    if (this->isGrowing) {
+        if (this->isGrowing != true) {
```
No, it's only ever set as 0 or 1 but I agree that it's suspicious. I was able to put it into a switch case form that looks a lot cleaner than before, with the two cases being true and false. Would it be better to bring it back to 0/1 since it's in a switch case? Or is true/false still fine

engineer124 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+                this->isActive = false;
+            }
+        }
+        return;
```
Like above, seems to be in the form of a switch-case for true/false

engineer124 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+        player->actor.posRot.pos.y = this->actor.posRot.pos.y; // y = 205.0f
+        player->actor.posRot.pos.z = this->actor.posRot.pos.z; // z = -1220.0f
+
+        rotY = this->actor.posRot.rot.y;
+        player->actor.shape.rot.y = rotY;
+        player->actor.posRot.rot.y = rotY;
+        player->currentYaw = rotY;
```
Well played

fig02 commented on spec:
```diff
@@ -2308,7 +2308,7 @@ endseg
     name "ovl_En_Fr"
     include "build/src/overlays/actors/ovl_En_Fr/z_en_fr.o"
-    include "build/data/overlays/actors/z_en_fr.data.o"
+    // include "build/data/overlays/actors/z_en_fr.data.o"
```
youll want to delete the line altogether, and then remove the asm file as well

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.h:
```diff
@@ -6,9 +6,57 @@
+    /* 01 */ FROG_BLUE,     // Front Left
+    /* 02 */ FROG_RED,      // Front Right
+    /* 03 */ FROG_PURPLE,   // Back Left
+    /* 04 */ FROG_WHITE     // Back Right
+} frogType;
```
struct and enum are written as `FrogType`

engineer124 replied:
I added two more enums, let me know if those are named properly

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -8,14 +9,169 @@ void EnFr_Init(Actor* thisx, GlobalContext* globalCtx);
-extern UNK_TYPE D_060011C0;
-extern UNK_TYPE D_06001534;
-extern UNK_TYPE D_0600B498;
+void EnFr_UpdateIdle(EnFr* this, GlobalContext* globalCtx);
+void EnFr_UpdateActive(EnFr* this, GlobalContext* globalCtx);
```
since these two functions get set to update directly, they are going to need to take `thisx` as their arg. and you can put them with the list of main functions above

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -8,14 +9,169 @@ void EnFr_Init(Actor* thisx, GlobalContext* globalCtx);
+void EnFr_UpdateIdle(EnFr* this, GlobalContext* globalCtx);
+void EnFr_UpdateActive(EnFr* this, GlobalContext* globalCtx);
+
+void EnFr_IsDivingIntoWater(EnFr* this, GlobalContext* globalCtx);
+void EnFr_DivingIntoWater(EnFr* this, GlobalContext* globalCtx);
```
we only need to forward declare functions when necessary. DivingToWater at least is not needed as the definition comes before any usages, should check the others as well

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -8,14 +9,169 @@ void EnFr_Init(Actor* thisx, GlobalContext* globalCtx);
+extern AnimationHeader D_060007BC;    // Frog Jumping
+extern AnimationHeader D_060011C0;    // Frog Landing
+
+//
+// Frogs actor.params system
```
you can use a multiline comment /* */ for something big like this

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -8,14 +9,169 @@ void EnFr_Init(Actor* thisx, GlobalContext* globalCtx);
+};
+
+// Flags for gSaveContext.eventChkInf[13]
+u16 sSongIndex[] = { 0x0002,   // ZL            (second bit)
+                     0x0004,   // Epona         (third bit)
+                     0x0010,   // Saria         (fifth bit)
+                     0x0008,   // Suns          (fourth bit)
+                     0x0020,   // SoT           (sixth bit)
+                     0x0040,   // Storms        (seventh bit)
+                     0x0001,   // Frogs 2 Song  (first bit)
+                     0x0000 }; // Padding
```
for multiline lists put a trailing comma on the last entry so the formatter doesnt do this weird indenting.
I also think the bit comments are a bit redundant, theyll prob become macros later anyway

engineer124 replied:
The bit comments are a bit excessive yeah. A lot of extra comments were really for my own benefit just to help with understanding the code lol. For the songs, would it be easy to get which bits represents which songs for mappings in general? I know I spent time on gz figuring out which bits/index/flags represents which songs/controller notes in general. Should I leave the comments in to help name the macros when that time comes? Or is that fairly easy without the comments? I have no preference either way.

fig02 replied:
whoever gets to audio stuff will take care of it, but you can leave the song comments for the time being to make it easier to reference until that point. i was talking specifically about `(first bit)` etc

fig02 replied:
also, the padding should be able to be omitted. the compiler should insert it on its own

engineer124 replied:
I'm also tempted to make an enum for the songs (mainly for sFrogToSongIndex). That might also work and be more readable, but feel free to object

fig02 replied:
should be fine. they may end up getting replaced by more general audio defines down the line, but since we dont know what those will be yet, we can figure it out later

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1A9A0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/EnFr_Init.s")
+static EnFrColor sEnFrColor[5] = { { 200, 170, 0, 255 },   // FROG_YELLOW
+                                   { 0, 170, 200, 255 },   // FROG_BLUE
```
this is just Color_RGBA8, no need for a frog specific struct

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+//      sLogSpotToFromWater[frog].xzDist is magnitude of xz distance frog travels
+//      sLogSpotToFromWater[frog].yaw is rot around y-axis of jumping back into water
+//      sLogSpotToFromWater[frog].yDist is change in y distance frog travels
+static LogSpotToFromWater sLogSpotToFromWater[6] = {
+    { 0.0, 0.0, 0.0 },              // Prop   (Where link pulls ocarina)
```
make sure to put f on the end of your floats

fig02 replied:
additionally, we try not to specify array lengths when possible. if people want to modify an array at a later time it would avoid throwing errors

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+                                12, // frames to play each remaining note
+                                0 };
+
+InitChainEntry sInitChain[] = {
+    ICHAIN_U8(unk_1F, 2, ICHAIN_CONTINUE),
```
make this static

engineer124 replied:
The reason I did not make this static was because it raised the error:
```
mips-linux-gnu-ld: build/src/overlays/actors/ovl_En_Fr/z_en_fr.o: in function `EnFr_Update':
(.text+0x2e0): undefined reference to `sInitChain'
mips-linux-gnu-ld: (.text+0x304): undefined reference to `sInitChain'
make: *** [Makefile:177: zelda_ocarina_mq_dbg.elf] Error 1
```

fig02 replied:
oh, is update a non matching? if so then makes sense, its a limitation of our asm preprocessor. in which case i recommend reverting this back to the D_ name until the function it matched

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+
+// targetScale (default = 150.0) Actor scale target for Math_SmoothScaleMaxF
+// Used as a frog grows from hearing a new child song
+static f32 sGrowingScale[] = { 180.0,   // 1st incremental scale
+                               210.0,   // 2nd incremental scale
```
f

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+                              3,   // C-Left
+                              4 }; // C-Up
+
+static s32 sDLists[] = { 0x060059A0, 0x06005BA0 };
+
```
dlists are type Gfx[], since this is an array of pointers you can do Gfx*

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+    this->actor.posRot.pos.y = sLogSpotToFromWater[this->actor.params].yDist + this->posLogSpot.y; // -80.0
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1AC34.s")
+    yaw = (s16)(sLogSpotToFromWater[this->actor.params].yaw * (32768.0f / M_PI)) + 32768;
 
+    this->actor.shape.rot.y = yaw;
+    this->actor.posRot.rot.y = yaw;
```
you should be able to remove this temp by doing `posRot.rot.y = shape.rot.y = (calculation)`

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+void EnFr_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFr* this = THIS;
+
+    // 6 Actor Instances for frogs:
+    //     -  5 NPC actor instances for the frogs themselves
+    //     -  1 Prop actor instance set to where Link plays Ocarina, manages 5 frogs
```
this info is already in the big comment block above right?

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+        this->actor.textId = 0x40AC;
+        this->actionFunc = EnFr_Idle;
+    } else {
+        // Translation: The argument is wrong!!
+        if ((this->actor.params >= 6) || (this->actor.params < 0)) {
+            osSyncPrintf(VT_COL(RED, WHITE));
+            osSyncPrintf("%s[%d] : 引数が間違っている！！(%d)\n", "../z_en_fr.c", 370, this->actor.params);
+            osSyncPrintf(VT_RST);
+            __assert("0", "../z_en_fr.c", 372);
+        }
+
+        this->objBankIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_GAMEPLAY_FIELD_KEEP);
+
+        // Translation: There is no bank!!
```
you can put the translation comments right above the string that its relevant for

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+}
+
+void EnFr_DrawIdle(EnFr* this) {
+    // Draw only purple frog when not actively listening to ocarina
+    if (this->actor.params != 4) {
```
you can use your frog enum value, and in which case the comment isnt really needed cause the enum tells you its for purple

engineer124 replied:
The issue is that params is shifted by one. So params 1-5 represents frogs 0-4. So the enum for purple frog is 3, and also the param for purple frog is 4.

petrie911 replied:
`if(this->actor.params = FROG_PURPLE + 1)` can be used then. Or even `if((this->actor.params - 1) == FROG_PURPLE)` if it matches.

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+
+#ifdef NON_MATCHING
+void EnFr_Update(Actor* thisx, GlobalContext* globalCtx) {
+
+    EnFr* this = THIS;
+    LightInfo* lightInfo;
+    s32 frogindex;
+    s16 randomNumber;
+
+    // if (globalCtx) {}
+
```
you can remove the linebreaks (except for one before the code starts) and prob take away the if(globalCtx)

engineer124 replied:
I left `if (globalCtx) {}` in there because it does seem to change the code in a meaningful way while trying to match. But perhaps I'll take it out and leave a comment that it's something to keep in mind

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+        sEnFrPointers.frogs[frogindex] = this;
+        Actor_ProcessInitChain(&this->actor, sInitChain);
+
+        // Skeleton for the frog
+        SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600B498, &D_06001534, &this->limbDrawTable,
+                         &this->transitionDrawTable, 24);
+        // Skeleton for the butterfly
```
these comments will be uneeded when the skeleton vars are named, but you can leave it for now if you want

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+
+        this->eyeTexIndex = 1;
+        randomNumber = ((s32)(Math_Rand_ZeroFloat(60.0f) + 20.0f) ) & 0xFFFF;
+        this->blinkTimer = (s16) randomNumber;
+        this->blinkFunc = EnFr_DecrementBlinkTimerUpdate;
```
can remove the space on the cast (and any more in the file if there are any)

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+}
+
+s32 EnFr_IsAboveAndWithin30xzDist(Player* player, EnFr* this) {
+    f32 xDistFromLink;
+    f32 zDistFromLink;
+    f32 yDistFromLink;
+
+    xDistFromLink = player->actor.posRot.pos.x - this->actor.posRot.pos.x;
```
you can do the assignment on the same line as the declare if you want, will make this more compact

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+
+void EnFr_InitJumpingOutOfWater(EnFr* this, GlobalContext* globalCtx) {
+    // Frogs jumping out one at a time in order sTimerJumpingOutOfWater
+    if (sEnFrPointers.flags == sTimerJumpingOutOfWater[this->actor.params - 1]) {
+
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060007BC, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_060007BC), 2, 0.0f);
+
+        EnFr_DrawActive(this);
+
+        this->actionFunc = EnFr_JumpingOutOfWater;
```
can prob remove all of the linebreaks here

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+}
+
+void EnFr_OrientOnLogSpot(EnFr* this, GlobalContext* globalCtx) {
+    s16 func;
+    func = Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 2, 10000, 100);
```
after you merge master youll see a comment on this function that describes the purpose of the return value. itd be better to name it after that, cause `func` can be pretty confusing.
also, this is a case where id assign with the declare to put it on the same line

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+    s16 func;
+    func = Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 2, 10000, 100);
+    this->actor.posRot.rot.y = this->actor.shape.rot.y;
+    if (func == 0 && this->skelAnime.animCurrentFrame == this->skelAnime.animFrameCount) {
+        sEnFrPointers.flags++;
```
if this ends up being a bool youll want !var

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+
+    if (this->skelAnime.animCurrentFrame == 6.0f) {
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+
+    } else if (this->skelAnime.animCurrentFrame == 3.0f) {
+        this->actor.gravity = -10.0f;
+        this->actor.velocity.y = 25.0f;
+
```
in general in this file id remove the line breaks between different blocks in the same if. i personally like doing so between different if blocks, but within the same one it makes it hard to follow i feel

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+}
+
+void EnFr_JumpingBackIntoWater(EnFr* this, GlobalContext* globalCtx) {
+    f32 y_underwater;
+
+    y_underwater = sLogSpotToFromWater[this->actor.params].yDist + this->posLogSpot.y;
```
camel case

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+}
+
+void EnFr_ButterflyPath(EnFr* this, GlobalContext* globalCtx) {
+    s16 rot_y;
+    f32 temp_sin;
```
no_snakes_allowed_homie

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+
+s32 EnFr_InitJumpingUp(EnFr* this, s32 frogIndex) {
+    EnFr* frog;
+    u8 audio_pitch;
+
```
last one ill comment on

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+void EnFr_Idle(EnFr* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    // If Link pulls out ocarina?
+    if (player->stateFlags2 & 0x2000000) {
```
yup. you can remove the comment if you want and itll be clear when we have a macro for the flags

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+
+    // Ocarina Song played
+    switch (globalCtx->msgCtx.unk_E3EE - 1) {
+        case 6: // Zeldas Lullaby   (unk_E3EE == 7)
+            this->songIndex = 0;
+            break;
+        case 5: // Epona's Song     (unk_E3EE == 6)
+            this->songIndex = 1;
+            break;
+        case 4: // Saria's Song     (unk_E3EE == 5)
```
if you add one to all of the cases you can remove the -1 from the switch check. this is just something the compiler did on its own to make things 0 indexed 
after which you can remove the comments

fig02 commented on src/overlays/actors/ovl_En_Fr/z_en_fr.c:
```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1CAAC.s")
+void EnFr_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s16 NoGlowRadius;
+    EnFr* this = THIS;
```
id just call this lightRadius. the NoGlow in the name refers to the fact that the light doesnt have a glow halo thing around it

fig02 reviewed (commented):
great job on this one, and sorry for the late review.
I may have more comments later, but i figure itd be good to get most things out of the way first.
youll want to merge master and grab the new math function names as well

fig02 commented:
the git diff is showing 2000+ changed files, so i think something happened there

engineer124 commented:
Yeah, I wasn't sure if that was normal or not, guess that confirms I did dun goof. It happened when I merged the main branch into my branch

engineer124 commented:
Not sure what went wrong or how to fix it

fig02 commented:
closing as a new pr has been made. refer to this closed pr for previous review comments
